operators/src/pro/executor/replay.rs

```diff
@@ -0,0 +1,736 @@
+            pending: Default::default(),
+            inner: self.inner.clone(),
+        })
+    }
+
+    pub fn try_send(&self, v: T) -> Result<(), SendError<T>> {
+        let mut lock = self.inner.lock().unwrap();
+        lock.try_send(v)
+    }
+}
+
+#[allow(clippy::missing_panics_doc)]
+impl<T> Sender<T>
+where
+    T: Clone + Unpin,
+{
+    pub fn send(&self, v: T) -> Send<T> {
+        let _ignore = self.pending.fetch_add(1, Ordering::Relaxed);
+        Send {
+            sender: self,
+            value: Some(v),
+        }
+    }
+
+    fn poll_send(
+        &self,
+        send: &mut Send<'_, T>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<(), SendError<T>>> {
+        let value = send.value.take().unwrap();
+        let mut lock = self.inner.lock().unwrap();
+        match lock.try_send(value) {
+            Ok(_) => Poll::Ready(Ok(())),
+            Err(SendError::Full(v)) => {
+                lock.update_send_waker(self.id, cx.waker());
+                let _ignore = send.value.insert(v);
+                Poll::Pending
+            }
+            Err(e) => Poll::Ready(Err(e)),
+        }
+    }
+
+    fn send_dropped(&self) {
+        if self.pending.fetch_sub(1, Ordering::Relaxed) == 1 {
+            let mut lock = self.inner.lock().unwrap();
+            lock.remove_send_waker(self.id);
+        }
+    }
+}
+
+#[allow(clippy::missing_panics_doc)]
+impl<T> Clone for Sender<T>
+where
+    T: Clone,
+{
+    fn clone(&self) -> Self {
+        let id = {
+            let mut lock = self.inner.lock().unwrap();
+            lock.create_sender()
+        };
+        Sender {
+            id,
+            pending: Default::default(),
+            inner: self.inner.clone(),
+        }
+    }
+}
+
+#[allow(clippy::missing_panics_doc)]
+impl<T> Drop for Sender<T>
+where
+    T: Clone,
+{
+    fn drop(&mut self) {
+        let mut lock = self.inner.lock().unwrap();
+        lock.remove_sender(self.id);
+    }
+}
+
+pub struct Send<'sender, T>
+where
+    T: Clone + Unpin,
+{
+    sender: &'sender Sender<T>,
+    value: Option<T>,
+}
+
+impl<T> Drop for Send<'_, T>
+where
+    T: Clone + Unpin,
+{
+    fn drop(&mut self) {
+        self.sender.send_dropped();
+    }
+}
+
+impl<T> Future for Send<'_, T>
+where
+    T: Clone + Unpin,
+{
+    type Output = Result<(), SendError<T>>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        self.sender.poll_send(self.get_mut(), cx)
+    }
+}
+
+// /////////////////////////////////////////////////////////////////////////////
+//
+// RECEIVER SIDE
+//
+// /////////////////////////////////////////////////////////////////////////////
+
+pub struct Receiver<T>
+where
+    T: Clone,
+{
+    id: u32,
+    pending: AtomicUsize,
+    inner: Arc<Mutex<Inner<T>>>,
+}
+
+#[allow(clippy::missing_panics_doc)]
+impl<T> Receiver<T>
+where
+    T: Clone,
+{
+    pub fn try_recv(&self) -> Result<T, ReceiveError> {
+        let mut lock = self.inner.lock().unwrap();
```

Comment:
I guess we have to see if lock poisoning can happen. If not, we have a util function that just ignores the error. I guess it is better than unwrapping wrt. panics.

Reply:
I don't see any case where poisoning could happen and used the util method now. b3cce3640346919b237496286858c0fef81403cd